src/herbivoresAndCarnivores.js

```diff
@@ -1,19 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden) {
+    super(name, health, hidden);
+  }
+  bite(grass) {
+    if (!grass.hidden && grass instanceof Herbivore) {
+      grass.health -= 50;
+
+      if (grass.health <= 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
[CODE PERFORMANCE] Instead of checking `grass.health` after each bite, you could simply filter the `Animal.alive` array for animals with health above 0 outside of the `bite` method. This way, you won't have to perform the health check and filter operation each time a bite occurs, which can improve performance since the filter will only be called when necessary.